charts/vmclarity/templates/apiserver/service-headless.yaml

```diff
@@ -0,0 +1,21 @@
+{{/*
+Headless Service does not include a ClusterIP so that the DNS record includes
+all Pod addresses. This makes it possible for the API Gateway to load balance
+all replicas instead of relying on kube-proxy to do it.
```

Comment:
how does the API GW is doing the LB?

Reply:
And why it is better than the k8s service lb algorithm?

Reply:
kube-proxy "loadbalancer" algorithm is fixed and it just picks a random pod from the list of Ready pods for the service. Using a headless service pushes this logic to something which can be smarter about it, for example nginx our current gateway can be configured to load balance round-robin (the default), by least number of connections, or by IP hash. https://docs.nginx.com/nginx/admin-guide/load-balancer/http-load-balancer/#choosing-a-load-balancing-method

Reply:
According to https://kodekloud.com/blog/clusterip-nodeport-loadbalancer/# k8s is also doing round-robin style for pods. But whatever to think is good.